DeviceAdministration/Infrastructure.UnitTests/Web/WebApiControllers/TelemetryApiControllerTests.cs

```diff
@@ -0,0 +1,191 @@
+{
+    public class TelemetryApiControllerTests
+    {
+        private readonly TelemetryApiController telemetryApiController;
+        private readonly Mock<IDeviceTelemetryLogic> telemetryLogic;
+        private readonly Mock<IAlertsLogic> alertLogic;
+        private readonly Mock<IDeviceLogic> deviceLogic;
+        private readonly Mock<IConfigurationProvider> configProvider;
+        private readonly IFixture fixture;
+
+        public TelemetryApiControllerTests()
+        {
+            this.telemetryLogic = new Mock<IDeviceTelemetryLogic>();
+            this.alertLogic = new Mock<IAlertsLogic>();
+            this.deviceLogic = new Mock<IDeviceLogic>();
+            this.configProvider = new Mock<IConfigurationProvider>();
+            this.telemetryApiController = new TelemetryApiController(this.telemetryLogic.Object,
+                                                                     this.alertLogic.Object,
+                                                                     this.deviceLogic.Object,
+                                                                     this.configProvider.Object);
+            this.telemetryApiController.InitializeRequest();
+            this.fixture = new Fixture();
+        }
+
+        [Fact]
+        public async void GetDashboardDevicePaneDataAsyncTest()
+        {
+            var deviceID = this.fixture.Create<string>();
+            var device = this.fixture.Create<DeviceModel>();
+            var telemetryFields = this.fixture.Create<IList<DeviceTelemetryFieldModel>>();
+            var telemetryModel = this.fixture.Create<IEnumerable<DeviceTelemetryModel>>();
+            var summModel = this.fixture.Create<DeviceTelemetrySummaryModel>();
+            this.deviceLogic.Setup(mock => mock.GetDeviceAsync(deviceID)).ReturnsAsync(device);
+            this.deviceLogic.Setup(mock => mock.ExtractTelemetry(device)).Returns(telemetryFields);
+
+            this.telemetryLogic.Setup(mock => mock.LoadLatestDeviceTelemetrySummaryAsync(deviceID, It.IsAny<DateTime>())).ReturnsAsync(summModel);
+            this.telemetryLogic.Setup(mock => mock.LoadLatestDeviceTelemetryAsync(deviceID, telemetryFields, It.IsAny<DateTime>()))
+                .ReturnsAsync(telemetryModel);
+
+            var res = await this.telemetryApiController.GetDashboardDevicePaneDataAsync(deviceID);
+            res.AssertOnError();
+            var data = res.ExtractContentAs<DashboardDevicePaneDataModel>();
+            Assert.Equal(data.DeviceTelemetryFields, telemetryFields.ToArray());
+            Assert.Equal(data.DeviceTelemetrySummaryModel, summModel);
+            Assert.Equal(data.DeviceTelemetryModels, telemetryModel.OrderBy(t => t.Timestamp).ToArray());
+
+            this.deviceLogic.Setup(mock => mock.ExtractTelemetry(device)).Throws(new Exception());
+            await Assert.ThrowsAsync<HttpResponseException>(() => this.telemetryApiController.GetDashboardDevicePaneDataAsync(deviceID));
```

Comment:
because of the line above it:
`this.deviceLogic.Setup(mock => mock.ExtractTelemetry(device)).Throws(new Exception()); `